File: src/main/java/es/urjc/daw04/model/dto/ProductDTO.java
package es.urjc.daw04.model.dto;

import java.util.List;

public record ProductDTO(
        Long id,
        String name,
        double price,
        String description,
        List<String> tags,
        double averageRating,
        CategoryDTO category,
        List<ProductReviewDTO> reviews
) {
}

File: frontend/app/api/dtos/product.dto.ts
import type { CategoryDTO } from "./category.dto";
import type { ImageDTO } from "./image.dto";
import type { ProductReviewDTO } from "./review.dto";

export interface ProductDTO {
  id: number;
  name: string;
  price: number;
  description: string;
  tags: string[];
  averageRating: number;
  category: CategoryDTO;
  images: ImageDTO[];
  reviews: ProductReviewDTO[];
}

export interface ProductSummaryDTO {
  id: number;
  name: string;
  price: number;
  description: string;
  tags: string[];
  averageRating: number;
  category: CategoryDTO;
  images: ImageDTO[];
}

export interface ProductCreateRequestDTO {
  name: string;
  price: number;
  description: string;
  tags: string[];
  categoryId: number;
}

export interface ProductUpdateRequestDTO {
  name: string;
  price: number;
  description: string;
  tags: string[];
  categoryId: number;
}
